docs/specification.md

```diff
@@ -370,30 +373,112 @@ when verification was in progress locally. If this is the case, the provisional
-keys. Further, maintainers may reset the states of the affected refs and sign
-new RSL entries with corrected states.
-
-TODO: evaluate if consecutive state verification fails on clients behind the
-times. Should recovery rewrite non-valid RSL entries? Defeats the purpose?
+## Recovery
+
+If every user were using gittuf and were performing each operation by
+generating all of the correct metadata, following the specification, etc., then
+the procedure for handling each situation is fairly straightforward. However,
+an important property of gittuf is to ensure that a malicious or erroneous
+party cannot make changes that impact the state of the repository in a negative
+manner. To address this, this section discusses how to handle situations where
+something has not gone according to protocol. The goal is to recover to a
+"known good" situation which does match the metadata which a set of valid
+gittuf clients would generate.
+
+### Recovery Mechanisms
+
+gittuf presents mechanisms that can be used in many different scenarios.
+
+#### M1: Reset to known good state
+
+This mechanism is utilized in scenarios where some change is rejected. For
+example, one or more commits may have been pushed to a branch that do not meet
+gittuf policy. The repository is updated such that these commits are neutralized
+and all Git refs match their latest RSL entries. This can take two forms:
+
+1. The rejected commit is removed and the state of the repo is set to the prior
+commit which is known to be good. This is used when all rejected commits are
+together at the end of the commit graph, making it easy to remove all of them.
+
+2. The rejected commit is _reverted_ where a new commit is introduced that
+reverses all the changes made in the reverted commit. This is needed when "good"
+commits that must be retained are interspersed with "bad" commits that must be
+rejected.
+
+In both cases, new RSL entries and annotations may be used to record the
+incident or to skip the invalid RSL entries corresponding to the rejected
+changes. TODO: should these new RSL entries come from authorized users for the
+affected namespace.
+
+#### M2: Create RSL entry on behalf of another user
+
+This mechanism is necessary for adoptions where a subset of developers do not
+use gittuf. When they submit changes to the main copy of the repository, they do
+not include RSL entries. Therefore, when a change is pushed to a branch by a
+non-gittuf user A, a gittuf user B can submit an RSL entry on their behalf.
+Additionally, the entry must identify the original user (TODO: cross-reference
+with signed pushes?) and include some evidence about why B thinks the change
+came from A (TODO: signed push can be evidence option?).
+
+TODO: decide if B needs to be authorized for the affected branch to sign for A.
+
+### Recovery Scenarios
+
+These scenarios are some examples where recovery is necessary.
+
+#### A change is made without an RSL entry
```

Comment:
What is the order of operations when it comes to an RSL entry and a commit?  Which comes first?  If it is RSL first, then we may also need to discuss handling the case where only the RSL entry appears.  This could occur with a legitimate client when it or the repo crashes / loses network connectivity in the middle of an action.

Reply:
The `Actor Workflows` section needs to be rewritten to describe these things: how RSL entries are created, how they are synced between repo copies and so on. For now, I've updated the "incorrect RSL entries" subsection to handle situations where the RSL entry appears without the corresponding change.

Reply:
Rewrote actor workflows above.